web/src/api/_reset-api-key/graphql/check-user-permission.graphql

```diff
@@ -0,0 +1,11 @@
+query CheckUserPermission($id: String!, $team_id: String!, $user_id: String!) {
```

Comment:
we seem to be recreating this query in multiple places, can we make it reusable? why not look into creating a centralized handler for all custom actions that are user-based that checks permissions and then routes the request to the relevant API function